### PR TITLE
Inspection audio warning. Fixes #23

### LINKED ISSUE #23: Option to verbally announce inspection elapsed time
Add an option to have an audio clip that plays to announce when 8s and 12s of inspection time have elapsed, if in inspection is enabled. This is a follow-up to #7.

### PR DESCRIPTION
**Pull request by marceloFA**
This PR adds a new configuration called "Use Inspection Time Audio Warning" which will play a simple beep when 8s and 12s of inspection time have been elapsed. 

I chose to create a draft first as there are some refinements to do before it can be proposed to merge.

@euphwes, would you help me with a small issue I'm having, please? 
The last time I contributed to a change in the app settings (#138), the new setting was automatically created for my dev user after I added it to ``app/persistence/settings_manager.py``. This time this didn't seem to have occurred and the new ``SettingCode.USE_INSPECTION_AUDIO_WARNING ``is not showing up on the ``settings_sections `` OrderedDict at ``app/routes/user/settings_routes.py``. I imagine this is somehow caused by my addition of the new setting into the ``affects`` list of ``SettingCode.USE_INSPECTION_TIME``. Am I missing something on how to implement a new setting that is affected by previously defined settings?

If you want to try out the changes of this PR in the /compete route timer, you can force
```javascript
 this.use_inspection_audio_warning = true;
``` 
instead of 
```javascript
app.userSettingsManager.get_setting(app.Settings.USE_INSPECTION_AUDIO_WARNING);
```
at ``timer_controller.js`` as the standard import of this setting won't currently work.

Thanks!

### CODE REVIEW

**Comment by euphwes** [comment]
Hey sorry I didn't get back to you on this yet. Had a busy day yesterday. I'll pull down your code so far and take a look, and point you in the right direction hopefully later today! Thanks for picking up this task!

**Comment by euphwes** [comment]
@marceloFA It looks like populating the setting is working properly for me:

1. Right after pulling your code, I visited the timer page. I checked the database for that setting for my user, and it existed (with the correct default `false` value).

2. I deleted that entry from the database. Visited the settings page and changed `Use Inspection Time Audio Warning` to `true` and submitted my change. Checked the database and confirmed the setting was there, with `true` value. Visited the timer page and let the timer run low, and heard the warning chimes after 8 and 12 seconds.

You said:
>as the standard import of this setting won't currently work.

In what way is that not working for you? You turn the setting on, but your timer still doesn't chime?

File: app/persistence/settings_manager.py [hunk 1]
```diff
@@ -193,7 +194,7 @@ def int_list_validator(value):
         validator     = boolean_validator,
         setting_type  = SettingType.BOOLEAN,
         default_value = FALSE_STR,
-        affects       = [SettingCode.HIDE_INSPECTION_TIME]
+        affects       = [SettingCode.HIDE_INSPECTION_TIME, SettingCode.USE_INSPECTION_AUDIO_WARNING]
```
Comment (euphwes) [inline app/persistence/settings_manager.py:197]: This looks right! You used it correctly.

**Review by euphwes — CHANGES_REQUESTED** [review]

File: app/static/js/timer/timer_controller.js [hunk 2]
```diff
@@ -396,10 +423,14 @@
...
     Timer.prototype._inspection_intervalFunction = function() {
+
         var inspection_elapsed_seconds = ((new Date()) - this.inspection_start_time).getSecondsFromMs();
         var seconds_remaining = this.INSPECTION_TIME_AMOUNT - inspection_elapsed_seconds;
 
-        if (seconds_remaining < this.AUTO_DNF_THRESHOLD) {
+        if ((inspection_elapsed_seconds >= this.EIGHT_SECS_ELAPSED) && this.use_inspection_audio_warning){
+            this._play_inspection_audio_warning(inspection_elapsed_seconds);
+        }
+        else if (seconds_remaining < this.AUTO_DNF_THRESHOLD) {
```
[the remark this answers is not in the document]
Comment (marceloFA) [reply app/static/js/timer/timer_controller.js:433]: Oh, now I see it! Thanks for this

**Comment by marceloFA** [comment]
Wes, It looks like populating the setting now is working properly for me too. Unfortunately, I couldn't notice what was going on before that make it didn't work. But at least everything in this PR code is functioning adequately now, in my point of view. 😂

**Pull request merged**